mat64/cholesky_test.go

```diff
@@ -210,6 +210,123 @@ func TestInverseCholesky(t *testing.T) {
 	}
 }
 
+func TestCholeskyRankOne(t *testing.T) {
+	rand.Seed(1)
+	for _, n := range []int{1, 2, 3, 4, 5, 7, 10, 20, 50, 100} {
+		for k := 0; k < 10; k++ {
+			data := make([]float64, n*n)
+			for i := range data {
+				data[i] = rand.NormFloat64()
+			}
+
+			var a SymDense
+			a.SymOuterK(1, NewDense(n, n, data))
+
+			xdata := make([]float64, n)
+			for i := range xdata {
+				xdata[i] = rand.NormFloat64()
+			}
+			x := NewVector(n, xdata)
+
+			var chol Cholesky
+			ok := chol.Factorize(&a)
+			if !ok {
+				t.Errorf("Bad random test, Cholesky factorization failed")
+				continue
+			}
+
+			alpha := rand.Float64()
```

Comment:
Doing factorization on random matrices with random negative alpha is very unpredictable, so here the tests use only alpha > 0.